deps/ccommon/src/channel/cc_tcp.c

```diff
@@ -397,6 +402,45 @@ tcp_reject(struct tcp_conn *sc)
     }
 }
 
+/*
+ * due to lack of a direct rejection API in POSIX, tcp_reject_all accepts
+ * connections ready on the listening socket, and immediately closes them.
+ * It does so until there are no more pending connections.
+ */
+void
+tcp_reject_all(struct tcp_conn *sc)
+{
+    int ret;
+    int sd;
+
+    for (;;) {
+        sd = accept(sc->sd, NULL, NULL);
+        if (sd < 0) {
+            if (errno == EINTR) {
```

Comment:
That is because `tcp_reject()` calls `_tcp_accept()` which has this logic. For actual accept we have to process one socket at a time (hence the `break` at the end of the `for` loop in `_tcp_accept()`, but if we are rejecting connections we don't need to worry about the application level setup and can tighten this loop, therefore `tcp_reject_all()` rewrites the accept for loop slightly to do just that.